src/lexer.jl

```diff
@@ -34,24 +34,25 @@ mutable struct Lexer{IO_t <: IO}
     charstore::IOBuffer
     current_char::Char
     doread::Bool
-    raw::Bool
 end
 
-Lexer(io, raw) = Lexer(io, position(io), 1, 1, position(io), 1, 1, position(io), Tokens.ERROR, IOBuffer(), ' ', false, raw)
+Lexer(io::IO_t, raw) where {IO_t} = Lexer{IO_t,raw ? RawToken : Token}(io, position(io), 1, 1, position(io), 1, 1, position(io), Tokens.ERROR, IOBuffer(), ' ', false)
```

Comment:
I think the second argument to the `Lexer` / `tokenize` has to be a type in order for everything to be type stable. So the constructor would be e.g. `Lexer("foo", Token)` or `Lexer("foo", RawToken)`.

Reply:
Doesn't appear to make a difference, but I'll change it

Reply:
Yes, I would assume the constant folding would work if you have `::Union{Type{Token}, Type{RawToken}}` and have a conditional on the type but perhaps you need two function definitions.